### DISCUSSION TITLE
How to extract a single Lap from Laps

### DISCUSSION THREAD

**Discussion by 3ric03** [theOehrly/Fast-F1#379] [Q&A]
Hi! I'd appreciate any help as I've been stuck on this for a while...

I'm trying to make a program that extracts the flying laps of the top 6 drivers in Q3 of quali. It's stored in a list of Lap, and then I pass the list into the Laps constructor

i have the following code:

`q1, q2, q3 = quali.laps.split_qualifying_sessions()
    top_drivers = quali.drivers[:6]
    
    hot_lap_list = list()
    for driver in top_drivers:
        hot_laps = q3.pick_driver(driver).pick_quicklaps()
        for lap in hot_laps.iterlaps():
            hot_lap_list.append(Lap(lap))
            
    sorted_hot_laps = Laps(hot_lap_list).sort_values(by='LapTime').reset_index(drop=True)`

but the last line always returns an error, it says it cannot find the "LapTime" key to sort. This is strange because I made sure all the elements in hot_lap_list are of the type "Lap". And they would all have the "LapTime" attribute. One of the issue I faced was changing the return value of .pick_quicklaps(), which is "Laps" to "Lap", I think I did it correct though?

This is the error message

> sorted_hot_laps = Laps(hot_lap_list).sort_values(by='LapTime').reset_index(drop=True)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pandas/core/frame.py", line 6768, in sort_values
    k = self._get_label_or_level_values(by, axis=axis)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pandas/core/generic.py", line 1778, in _get_label_or_level_values
    raise KeyError(key)
KeyError: 'LapTime'<

Thank you in advance for any help

**Comment by theOehrly** [comment]
```
q1, q2, q3 = quali.laps.split_qualifying_sessions()
top_drivers = quali.drivers[:6]

hot_laps = q3[q3['DriverNumber'].isin(top_drivers)].pick_quicklaps()

sorted_hot_laps = hot_laps.sort_values(by='LapTime').reset_index(drop=True)
```

This should work? A bit of pandas indexer magic.

There'll soon be a `pick_drivers` function that lets you select multiple drivers at once. Then you'll be able to do this more easily.
